source/common/upstream/cds_api_impl.cc

```diff
@@ -88,7 +88,7 @@ void CdsApiImpl::onConfigUpdate(const ResourceVector& resources, const std::stri
         ENVOY_LOG(debug, "cds: add/update cluster '{}'", cluster_name);
       }
     } catch (const EnvoyException& e) {
-      exception_msgs.push_back(e.what());
+      exception_msgs.push_back(fmt::format("{}: {}", cluster_name, e.what()));
```

Comment:
TBH, I think this is all getting a little crazy. We're iterating over human readable strings, which might ultimately be parsed out by machines, see https://github.com/envoyproxy/envoy/issues/6026.

Ultimately what we want is https://github.com/envoyproxy/envoy/issues/5739. In the interim, I think we should have in both of these cases some proto structure resembling this, and we can spit out the JSON equivalent in the exception messages etc.

This probably needs a broader discussion, but we should be very careful about not forming strong dependencies on non-structured text error messages that are supposed to be indicating something that needs to be parsed again, this is an anti-pattern </rant>

CC @mattklein123 @fredlas @SecurityInsanity